### PR TITLE
perf(ext): extend minimum token freshness

### PR DESCRIPTION
**Pull request by fa-sharp**
- Extends the minimum token freshness to 20 minutes to reduce `/api/token` API calls from the extension
- Centralizes the token stale/fresh check in the extension, so staleness is always checked before calling `/api/token`

### CODE REVIEW

File: extension/src/lib/state/auth.ts [hunk 1]
```diff
@@ -43,10 +43,15 @@ export class AuthManager {
```
Comment (farshybot) [inline extension/src/lib/state/auth.ts:43]: This 1-minute buffer between the extension (19 min stale threshold) and server (20 min freshness threshold) is a reasonable conservative approach to avoid edge-case races. Consider adding a comment here explaining this buffer.

File: web/test/token.test.ts [hunk 2]
```diff
@@ -48,7 +48,7 @@ describe("API: /token routes", () => {
     const currentAuthToken = app.crypto.encryptTokenData({
       accessToken: "current-access",
       refreshToken: "current-refresh",
-      expires: Date.now() + 10 * 60 * 1000, // active for 10 minutes
+      expires: Date.now() + 30 * 60 * 1000, // active for 30 minutes
```
Comment (farshybot) [inline web/test/token.test.ts:51]: The test correctly uses a 30-minute token (well above the 20-minute threshold) to verify the non-refresh path.

**Review by farshybot — APPROVED** [review]
## PR Review: Extend minimum token freshness (#207)

### Summary
This PR reduces `/api/token` API calls by extending the token freshness window from ~5 minutes to 20 minutes, and adds a client-side staleness check in the extension to skip API calls entirely when the cached token is still fresh.

### What this PR does well:
1. **Centralized staleness check** — The extension now checks token freshness before making any API call, avoiding unnecessary network requests entirely when the cached token is valid.
2. **Aligned server threshold** — The server's token freshness threshold was raised to 20 minutes to match the extension's new behavior, ensuring consistency between client and server logic.
3. **Clear naming** — Renaming `EXPIRE_GRACE_MILLIS` → `TOKEN_FRESH_MILLIS` and `TOKEN_STALE_TIME` improves code readability.
4. **Deduplication preserved** — The `fetchTokenInFlight` pattern continues to prevent concurrent requests.

### Changes reviewed:
| File | Change |
|------|--------|
| `extension/src/lib/state/auth.ts` | `TOKEN_STALE_TIME`: 4min → 19min; `fetchToken()` is now async with early return for fresh tokens |
| `web/server/routes/token/index.ts` | `TOKEN_FRESH_MILLIS`: 5min → 20min grace period |
| `web/test/token.test.ts` | Test token expiry: 10min → 30min |
| `extension/src/entrypoints/background.ts` | Minor comment cleanup |

### Edge cases & considerations:

1. **1-minute buffer between extension and server** — The extension marks tokens stale at 19 minutes, but the server only returns the existing token if it has >20 minutes remaining. This means there's a 1-minute window where the extension thinks a token is fresh but the server would refresh it anyway. This is a reasonable conservative buffer and avoids edge-case races.

2. **Return type change is backward-compatible** — `fetchToken` went from sync to async, but callers (the `useAuth` hook via `sendMessage`) handle promises correctly, so this is fine.

3. **Test coverage** — The test for "does not refresh tokens when not expired" now uses a 30-minute token, which correctly exercises the fresh path.

### Verdict
**APPROVE** — The changes are correct, well-scoped, and achieve the stated performance goal of reducing `/api/token` API calls. The centralized staleness check is a clean improvement.
